### PR TITLE
refactor(algo_wrapper, configs): rename update cycle and refactor structure

### PR DESCRIPTION
**Pull request by Gaiejj**
## Description

Rename update cycle and refactor structure.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/OmniSafeAI/omnisafe/blob/HEAD/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: omnisafe/common/experiment_grid.py [hunk 1]
```diff
@@ -338,37 +338,22 @@ def variants(self):
         """
         flat_variants = self._variants(self.keys, self.vals)
 
-        def unflatten_var(var):
+        def remove_duplicate(var):
```
Comment (muchvo) [inline omnisafe/common/experiment_grid.py:341]:
```suggestion
        def check_duplicate(var):
```

File: omnisafe/common/experiment_grid.py [hunk 2]
```diff
@@ -338,37 +338,22 @@ def variants(self):
...
-                else:
-                    assert key not in new_var, "You can't assign multiple values to the same key."
-                    new_var[key] = value
+                assert key not in new_var, "You can't assign multiple values to the same key."
+                new_var[key] = value
 
             # make sure to fill out the nested dict.
             for key in unflatten_set:
-                new_var[key] = unflatten_var(new_var[key])
+                new_var[key] = remove_duplicate(new_var[key])
```
Comment (muchvo) [inline omnisafe/common/experiment_grid.py:352]:
```suggestion
                new_var[key] = check_duplicate(new_var[key])
```

File: omnisafe/common/experiment_grid.py [hunk 3]
```diff
@@ -338,37 +338,22 @@ def variants(self):
...
 
             return new_var
 
-        return [unflatten_var(var) for var in flat_variants]
+        return [remove_duplicate(var) for var in flat_variants]
```
Comment (muchvo) [inline omnisafe/common/experiment_grid.py:356]:
```suggestion
        return [check_duplicate(var) for var in flat_variants]
```

File: omnisafe/common/logger.py [hunk 4]
```diff
@@ -283,7 +286,10 @@ def dump_tabular(self) -> None:
             key_lens = list(map(len, self._current_row.keys()))
             max_key_len = max(15, *key_lens)
             for key, val in self._current_row.items():
-                table.add_row(key[:max_key_len], str(val)[:max_key_len])
+                if self._headers_minmax[key]:
+                    table.add_row(f'{key}/Mean', str(val)[:max_key_len])
```
Comment (muchvo) [inline omnisafe/common/logger.py:290]:
```suggestion
                    table.add_row(f'{key[:max_key_len]}/Mean', str(val)[:max_key_len])
```

File: omnisafe/common/statistics_tools.py [hunk 5]
```diff
@@ -312,37 +312,22 @@ def variants(self, keys, vals):
         """
         flat_variants = self._variants(keys, vals)
 
-        def unflatten_var(var):
+        def remove_duplicate(var):
```
Comment (muchvo) [inline omnisafe/common/statistics_tools.py:315]:
```suggestion
        def check_duplicate(var):
```

File: omnisafe/common/statistics_tools.py [hunk 6]
```diff
@@ -312,37 +312,22 @@ def variants(self, keys, vals):
...
-                else:
-                    assert key not in new_var, "You can't assign multiple values to the same key."
-                    new_var[key] = value
+                assert key not in new_var, "You can't assign multiple values to the same key."
+                new_var[key] = value
 
             # make sure to fill out the nested dict.
             for key in unflatten_set:
-                new_var[key] = unflatten_var(new_var[key])
+                new_var[key] = remove_duplicate(new_var[key])
```
Comment (muchvo) [inline omnisafe/common/statistics_tools.py:326]:
```suggestion
                new_var[key] = check_duplicate(new_var[key])
```

File: omnisafe/common/statistics_tools.py [hunk 7]
```diff
@@ -312,37 +312,22 @@ def variants(self, keys, vals):
...
 
             return new_var
 
-        return [unflatten_var(var) for var in flat_variants]
+        return [remove_duplicate(var) for var in flat_variants]
```
Comment (muchvo) [inline omnisafe/common/statistics_tools.py:330]:
```suggestion
        return [check_duplicate(var) for var in flat_variants]
```

File: omnisafe/algorithms/on_policy/second_order/cpo.py [hunk 8]
```diff
@@ -289,7 +282,8 @@ def _update_actor(
 
             def project(data: torch.Tensor, low: float, high: float) -> torch.Tensor:
                 """Project data to [low, high] interval."""
-                return torch.max(torch.min(data, torch.tensor(high)), torch.tensor(low))
+                return torch.clamp(data, low, high)
+                # return torch.max(torch.min(data, torch.tensor(high)), torch.tensor(low))
```
Comment (zmsn-2077) [inline omnisafe/algorithms/on_policy/second_order/cpo.py:286]: delete this line

File: omnisafe/algorithms/on_policy/second_order/cpo.py [hunk 9]
```diff
@@ -331,6 +325,70 @@ def f_b(lam):
             nu_star = torch.sqrt(2 * self._cfgs.algo_cfgs.target_kl / (s + 1e-8))
             step_direction = -nu_star * p
 
+        return step_direction, lambda_star, nu_star
+
+    # pylint: disable=invalid-name,too-many-arguments,too-many-locals
```
Comment (zmsn-2077) [inline omnisafe/algorithms/on_policy/second_order/cpo.py:330]: disable-next

**Review by zmsn-2077 — CHANGES_REQUESTED** [review]
I think the version can be bumped to 0.4.0. Offline and model-based will be bumped to another version.

**Review by zmsn-2077 — APPROVED** [review]
LGTM.

**Review by muchvo — APPROVED** [review]
LGTM.

**Pull request merged**